stripe/src/main/java/com/stripe/android/net/PollingNetworkHandler.java

```diff
@@ -0,0 +1,188 @@
+package com.stripe.android.net;
+
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.Looper;
+import android.os.Message;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.stripe.android.exception.PollingFailedException;
+import com.stripe.android.exception.StripeException;
+import com.stripe.android.model.Source;
+import com.stripe.android.model.SourceRedirect;
+
+/**
+ * Class to handle polling on a background thread.
+ */
+class PollingNetworkHandler {
+
+    private static final String AUTHORIZATION_FAILED = "The redirect authorization request failed.";
+    private static final long DEFAULT_TIMEOUT_MS = 10000L;
+    private static final int INITIAL_DELAY_MS = 100;
+    private static final String POLLING_EXPIRED = "The polling request has expired.";
+    private static final int POLLING_MULTIPLIER = 2;
+
+    private static final int SUCCESS = 1;
+    private static final int PENDING = 2;
+    private static final int FAILURE = 3;
+    private static final int ERROR = -1;
+    private static final int EXPIRED = -2;
+
+    private final String mClientSecret;
+    private final String mPublishableKey;
+    private final String mSourceId;
+
+    private final long mTimeoutMs;
+    private final boolean mIsInSingleThreadMode;
+    @NonNull private final SourceRetriever mSourceRetriever;
+
+    private Handler mNetworkHandler;
+    private Handler mUiHandler;
+
+    private final Runnable pollRunnable = new Runnable() {
+        @Override
+        public void run() {
+            Message message = null;
+            try {
+                Source source = mSourceRetriever.retrieveSource(
+                        mSourceId,
+                        mClientSecret,
+                        mPublishableKey);
+                if (source.getRedirect() != null) {
```

Comment:
Ok -- I thought the check was in the redirect object. Should we look at that at all?